cargo-pgx/src/command/cross_options.rs

```diff
@@ -0,0 +1,51 @@
+use clap::Args;
+use std::path::PathBuf;
+
+#[derive(Args, Clone, Default, Debug)]
+pub(crate) struct CrossBuildArgs {
+    /// Cross-compilation target - passing this option will make cargo-pgx assume we are cross-compiling
+    #[clap(long)]
+    pub(crate) target: Option<String>,
+    /// Cross-compilation sysroot
+    #[clap(long)]
+    pub(crate) sysroot: Option<PathBuf>,
+    /// Host sysroot
+    #[clap(long)]
+    pub(crate) host_sysroot: Option<PathBuf>,
+    /// Host pg_config
+    #[clap(long)]
+    pub(crate) host_pg_config: Option<PathBuf>,
+}
+
+impl CrossBuildArgs {
+    pub(crate) fn is_cross_compiling(&self) -> bool {
+        self.target.is_some()
+    }
+
+    pub(crate) fn to_build(&self) -> CrossBuild {
+        match self.is_cross_compiling() {
+            true => CrossBuild::Target {
+                target: self.target.clone().unwrap(),
+                sysroot: self.sysroot.clone(),
+            },
+            false => CrossBuild::None,
+        }
+    }
+
+    pub(crate) fn to_host_build(&self) -> CrossBuild {
+        match self.is_cross_compiling() {
```

Comment:
µnit: match on bool isn't particularly idiomatic -- `if` is better (here and elsewhere).

Reply:
Fair, I got into the stupid habit because in some places they're more concise and (to me) more readable. Because `rustfmt` enforces new lines in `if {} else{}` blocks, and I wanted something like. Here it doesn't make sense, as you say.

```rust
if (answer) { foo.answer(42); }
else { foo.ignore(); }
```

Reply:
> use_small_heuristics = "Max"

This line in our rustfmt.toml means that rustfmt will only enforce that if it cannot actually one-line your code. Otherwise it will attempt to put a simple if-else on one line. It will never do so to a `match`. So you actually are defeating rustfmt's attempt to make your code the smallest.